agent/task.go

```diff
@@ -203,6 +210,12 @@ func (tm *taskManager) run(ctx context.Context) {
 			if cancel != nil {
 				// cancel outstanding operation.
 				cancel()
```

Comment:
I see. So, it sounds like the bug is in the context handling. #1159 works because it _completely_ disables the cancellation.

This is a race condition in the docker daemon. I'm not sure how to deal with this without doing a retry loop. Even then, if we don't have a guarantee about the state of the container after the `Create` method returns, we'd have to keep that container ID around until we can be _absolutely_ sure.

We'll really need to fix this in the docker daemon, such that one can remove a container that is in the process of being created.

I've filed docker/docker#24858.

This PR should make it so `Remove` and `container.create` aren't issued concurrently, so it is still needed.